eclair-core/src/test/scala/fr/acinq/eclair/api/JsonRpcServiceSpec.scala

```diff
@@ -0,0 +1,241 @@
+          state = "CONNECTED",
+          address = Some(Alice.nodeParams.publicAddresses.head),
+          channels = 1)
+      }
+    }))
+
+    val mockBobPeer = system.actorOf(Props(new {} with MockActor {
+      override def receive = {
+        case GetPeerInfo => sender() ! PeerInfo(
+          nodeId = Bob.nodeParams.nodeId,
+          state = "DISCONNECTED",
+          address = None,
+          channels = 1)
+      }
+    }))
+
+
+    val mockService = new MockService(defaultMockKit.copy(
+      switchboard = system.actorOf(Props(new {} with MockActor {
+        override def receive = {
+          case 'peers => sender() ! Map(Alice.nodeParams.nodeId -> mockAlicePeer, Bob.nodeParams.nodeId -> mockBobPeer)
+        }
+      }))
+    ))
+    
+    import mockService.formats
+    import mockService.serialization
+    
+    val postBody = JsonRPCBody(method = "peers", params = Seq.empty)
+    
+    Post("/", postBody) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      addHeader("Content-Type", "application/json") ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val response = entityAs[JsonRPCRes]
+        val peerInfos = response.result.asInstanceOf[Seq[Map[String,String]]]
+        assert(peerInfos.size == 2)
+        assert(peerInfos.head.get("nodeId") == Some(Alice.nodeParams.nodeId.toString))
+        assert(peerInfos.head.get("state") == Some("CONNECTED"))
+        matchTestJson("peers", false, response)
+      }
+  }
+
+  test("GetInfo response should include this node ID") {
+    val mockService = new {} with MockService {
+      override def getInfoResponse: Future[GetInfoResponse] = Future.successful(GetInfoResponse(
+        nodeId = Alice.nodeParams.nodeId,
+        alias = Alice.nodeParams.alias,
+        port = 9735,
+        chainHash = Alice.nodeParams.chainHash,
+        blockHeight = 123456
+      ))
+    }
+    import mockService.formats
+    import mockService.serialization
+
+    val postBody = JsonRPCBody(method = "getinfo", params = Seq.empty)
+
+    Post("/", postBody) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      addHeader("Content-Type", "application/json") ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val resp = entityAs[JsonRPCRes]
+        assert(resp.result.toString.contains(Alice.nodeParams.nodeId.toString))
+        matchTestJson("getinfo", false ,resp)
+      }
+  }
+
+
+  private def readFileAsString(path: Path): String = Files.exists(path) match {
+    case true => new String(Files.readAllBytes(path.toAbsolutePath))
+    case false => throw new IllegalArgumentException(s"Mock file for $path does not exist, please use 'overWrite' first.")
+  }
+
+  private def matchTestJson(rpcMethod: String, overWrite: Boolean, response: JsonRPCRes)(implicit formats: Formats) = {
```

Comment:
This is the function used to match the api response with the mock, if the developer changing the API ackowledges the breaking change he/she can set "overWrite" to true, to let the test over write the mock data. If overWrite = true the test will fail still to prevent forgetting it to true (which would defeat the purpose of the test)